CondTools/SiPhase2Tracker/plugins/DTCCablingMapProducer.cc

```diff
@@ -97,28 +99,44 @@ class DTCCablingMapProducer : public edm::one::EDAnalyzer<> {
   edm::ParameterSetDescription desc;
   desc.setComment("Stores a TrackerDetToDTCELinkCablingMap object into the database from a CSV file.");
-  desc.add<bool>("generate_fake_valid_gbtlink_and_elinkid", false);
+  desc.add<std::string>("dummy_fill_mode", "DUMMY_FILL_DISABLED");
   desc.add<int>("verbosity", 0);
-  desc.add<unsigned>("csvFormat_idetid", 0);
   desc.add<unsigned>("csvFormat_ncolumns", 15);
-  desc.add<unsigned>("csvFormat_idtcid", 10);
-  desc.add<unsigned>("csvFormat_igbtlinkid", 11);
-  desc.add<unsigned>("csvFormat_ielinkid", 12);
+  desc.add<unsigned>("csvFormat_idetid", 0);
+  desc.add<unsigned>("csvFormat_idtcid", 0);
+  desc.add<unsigned>("csvFormat_igbtlinkid", 0);
+  desc.add<unsigned>("csvFormat_ielinkid", 0);
   desc.add<long long unsigned int>("iovBeginTime", 1);
   desc.add<std::string>("record", "TrackerDTCCablingMapRcd");
   desc.add<std::vector<std::string>>("modulesToDTCCablingCSVFileNames", std::vector<std::string>());
   descriptions.add("DTCCablingMapProducer", desc);
 }
 
 DTCCablingMapProducer::DTCCablingMapProducer(const edm::ParameterSet& iConfig)
-    : generate_fake_valid_gbtlink_and_elinkid_(iConfig.getParameter<bool>("generate_fake_valid_gbtlink_and_elinkid")),
-      verbosity_(iConfig.getParameter<int>("verbosity")),
-      csvFormat_ncolumns_(iConfig.getParameter<unsigned>("csvFormat_ncolumns")),
-      csvFormat_idetid_(iConfig.getParameter<unsigned>("csvFormat_idetid")),
-      csvFormat_idtcid_(iConfig.getParameter<unsigned>("csvFormat_idtcid")),
+    : verbosity_(iConfig.getParameter<int>("verbosity")),
+      csvFormat_ncolumns_  (iConfig.getParameter<unsigned>("csvFormat_ncolumns")),
+      csvFormat_idetid_    (iConfig.getParameter<unsigned>("csvFormat_idetid")),
+      csvFormat_idtcid_    (iConfig.getParameter<unsigned>("csvFormat_idtcid")),
+      csvFormat_igbtlinkid_(iConfig.getParameter<unsigned>("csvFormat_igbtlinkid")),
+      csvFormat_ielinkid_  (iConfig.getParameter<unsigned>("csvFormat_ielinkid")),
       iovBeginTime_(iConfig.getParameter<long long unsigned int>("iovBeginTime")),
       pCablingMap_(std::make_unique<TrackerDetToDTCELinkCablingMap>()),
       record_(iConfig.getParameter<std::string>("record")) {
+  std::string const dummy_fill_mode_param = iConfig.getParameter<std::string>("dummy_fill_mode");
+  
+  // We pass from the easy to use string to an int representation for this mode flag, as it is more efficient in comparisons
+  if (dummy_fill_mode_param == "DUMMY_FILL_DISABLED")
+    dummy_fill_mode_ = DUMMY_FILL_DISABLED;
+  else if (dummy_fill_mode_param == "DUMMY_FILL_ELINK_ID")
+    dummy_fill_mode_ = DUMMY_FILL_ELINK_ID;
+  else if (dummy_fill_mode_param == "DUMMY_FILL_ELINK_ID_AND_GBT_ID")
+    dummy_fill_mode_ = DUMMY_FILL_ELINK_ID_AND_GBT_ID;
+  else {
+    std::ostringstream message;
+    message << "Parameter dummy_fill_mode with invalid value: " << dummy_fill_mode_param;
+    throw cms::Exception(message.str());
```

Comment:
this should be:
```
throw cms::Exception("InvalidDummyFillMode") << "Parameter dummy_fill_mode with invalid value: " << dummy_fill_mode_param;
```
the value used in the exception constructor is the category, not the message.

Reply:
Fixed, also in other parts of the code, to specify the category properly.